app/src/components/governance/PageProposal.vue

```diff
@@ -1,168 +1,41 @@
-        </div>
-      </div>
+    <div v-else class="proposal">
+      <ProposalHeader
+        :title="proposal.title"
+        :type="proposal.type"
+        :proposer="proposal.proposer.address"
+        :status="status"
+        @open-vote-modal="onVote"
+        @open-deposit-modal="onDeposit"
+      />
 
-      <TextBlock :content="proposal.description" />
+      <ProposalStatusBar
+        v-if="tallyHasValues"
+        :status="status.badge"
+        :status-begin-time="new Date(proposal.statusBeginTime).toUTCString()"
+        :total-votes="proposal.tally.total"
+        :proposal-id="proposal.proposalId"
+        :vote-percentage="proposal.tally.totalVotedPercentage"
+        :yes-votes="Number(proposal.tally.yes)"
+        :no-votes="Number(proposal.tally.no)"
+        :veto-votes="Number(proposal.tally.veto)"
+        :abstain-votes="Number(proposal.tally.abstain)"
+        :deposit-total="proposal.detailedVotes.depositsSum"
+      />
 
-      <ul v-if="proposal.status === 'DepositPeriod'" class="row">
-        <li>
-          <h4>Deposit Count</h4>
-          <span>
-            {{ proposal.deposit }}
-            /
-            {{ parameters.depositThreshold }}
-            {{ parameters.depositDenom || currentNetwork.stakingDenom }}
-          </span>
-        </li>
-      </ul>
+      <ParticipantList v-if="participants" :participants="participants" />
 
-      <ul v-if="proposal.status !== `DepositPeriod`" class="row">
-        <li v-if="proposal.status === `VotingPeriod`">
-          <h4>Total Vote Count</h4>
-          <span>
-            {{ proposal.tally.totalVotedPercentage | percent }} /
-            {{ proposal.tally.total | prettyInt }}
-          </span>
-        </li>
-        <li>
-          <h4>Yes</h4>
-          <span>
-            {{
-              noVotes
-                ? 0
-                : (proposal.tally.yes / proposal.tally.total) | percent
-            }}
-            /
-            {{ proposal.tally.yes | prettyInt }}
-          </span>
-        </li>
-        <li>
-          <h4>No</h4>
-          <span>
-            {{
-              noVotes ? 0 : (proposal.tally.no / proposal.tally.total) | percent
-            }}
-            /
-            {{ proposal.tally.no | prettyInt }}
-          </span>
-        </li>
-        <li>
-          <h4>Veto</h4>
-          <span>
-            {{
-              noVotes
-                ? 0
-                : (proposal.tally.veto / proposal.tally.total) | percent
-            }}
-            /
-            {{ proposal.tally.veto | prettyInt }}
-          </span>
-        </li>
-        <li>
-          <h4>Abstain</h4>
-          <span>
-            {{
-              noVotes
-                ? 0
-                : (proposal.tally.abstain / proposal.tally.total) | percent
-            }}
-            /
-            {{ proposal.tally.abstain | prettyInt }}
-          </span>
-        </li>
-      </ul>
+      <template v-if="proposal.detailedVotes.timeline">
```

Comment:
```suggestion
      <template v-if="proposal.detailedVotes.timeline && proposal.detailedVotes.timeline.length > 0">
```

Reply:
timeline is an Object. Then it should be `Object.keys(proposal.detailedVotes.timeline).length > 0`